py3status/modules/deadbeef.py

```diff
@@ -67,56 +85,40 @@ def _is_running(self):
 
     def deadbeef(self):
         color = self.color_stopped
-        artist = isplaying = length = playback_time = title = \
-            tracknumber = year = ""
+        status = None
 
         if not self._is_running():
             return {
                 'cached_until': self.py3.time_in(self.cache_timeout),
-                'color': color,
-                'full_text': ''
+                'full_text': self.py3.safe_format(self.format),
+                'color': color
             }
 
-        # mix metadata fields in with the delimiters
-        fmt = self.delimiter.join([
-            '%artist%', '%isplaying%', '%length%', '%playback_time%',
-            '%title%', '%tracknumber%', '%year%'
-        ])
         # Starting deadbeef may generate lot of startup noises either with
-        # or without error codes. Running deadbeef command below may sometimes
-        # change how things behaves onscreen. We use subprocess to ignore error
-        # codes and using pgrep to control how status output and color will
-        # look... mainly to build consistency better between versions.
-
-        # run command and get output
-        out = subprocess.check_output(['deadbeef', '--nowplaying-tf', fmt])
+        # or without error codes. Running command below may sometimes change
+        # how things behaves onscreen. We use subprocess to ignore error codes.
+        # We use pgrep and hidden placeholders to dictate how status output
+        # and color should look... mainly to stay consistency between versions.
+        out = check_output('deadbeef --nowplaying-tf "%s"' % self.fmt, shell=True)
         out = out.decode('utf-8')
 
-        # get metadata results out of the string using the delimiters
+        # 7.1 backward compatible
         if out != 'nothing':
```

Comment:
Yes, literal `nothing` only in `7.1`.
`7.0` and `7.2` don't return anything back.